File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/engine/TouchMessagePool.java
/**
 * 
 */
package com.abubusoft.xenon.engine;

import com.abubusoft.xenon.core.collections.ObjectPool;

/**
 * @author Francesco Benincasa
 *
 */
public class TouchMessagePool extends ObjectPool<TouchMessage> {

	public TouchMessagePool(int maxSize) {
		super(maxSize);
	}
	
	public TouchMessagePool() {
		super(ObjectPool.DEFAULT_CAPACITY);
	}

	@Override
	protected TouchMessage createPooledObject() {
		return new TouchMessage();
	}

}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/engine/TouchMessagePool.kt
/**
 *
 */
package com.abubusoft.xenon.engine

import com.abubusoft.xenon.core.collections.ObjectPool

/**
 * @author Francesco Benincasa
 */
class TouchMessagePool : ObjectPool<TouchMessage?> {
    constructor(maxSize: Int) : super(maxSize) {}
    constructor() : super(DEFAULT_CAPACITY) {}

    public override fun createPooledObject(): TouchMessage {
        return TouchMessage()
    }
}